src/pages/blocks/[district].tsx

```diff
@@ -0,0 +1,244 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Box,
+  Typography,
+  TextField,
+  Button,
+  InputAdornment,
+} from '@mui/material';
+import Header from '../../components/Header';
+import { useTheme } from '@mui/material/styles';
+import { useTranslation } from 'next-i18next';
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
+import Search from '@mui/icons-material/Search';
+import { ArrowDropDown, KeyboardArrowRight } from '@mui/icons-material';
+import AddIcon from '@mui/icons-material/Add';
+import { useRouter } from 'next/router';
+import { GetStaticPaths } from 'next';
+import CreateBlockModal from './components/CreateBlockModal';
+import FilterModal from './components/FilterModal';
+
+interface BlocksProps {
+  // Define any props if needed
+}
+
+const Blocks: React.FC<BlocksProps> = () => {
+  const { t } = useTranslation();
+  const theme = useTheme<any>();
+  const router = useRouter();
+  const { district } = router.query;
+
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const [openModal, setOpenModal] = useState(false); // State for modal
+  const [filterModalOpen, setFilterModalOpen] = useState(false); // State for filter modal
+  const [searchInput, setSearchInput] = useState(''); // State for search input
+  const [blocks, setBlocks] = useState([
+    'Bhivapur',
+    'Kamptee',
```

Comment:
@Aar-if please hard coding blocks here, instead create one block service, write method to `getBlocks`. Use this hardcoded blocks there and use that method here.